docs/guide/templates.md

```diff
@@ -192,3 +192,15 @@ class MyComponent < ViewComponent::Base
   strip_trailing_whitespace(false)
 end
 ```
+
+## Setting template file extension
+
+Sometimes a component will warrant the use of multiple file types for a single template handler. An example would be ERB handled files for both HTML and CSS. By default ViewComponent will return a "More than one template found for" error. To avoid this set the `VC_TEMPLATE_EXTENSION` constant in your component to the file extension of the template you wish your component to use.
```

Comment:
Is there a reason we wouldn't want to change the existing logic to look for multiple extensions and unique by that?

Reply:
@BlakeWilliams I'm not clear on why a component would need multiple template file extensions. Could you provide an example to help me understand?

My understanding and experience is that a component renders a single type of output, usually HTML. So the variant rules apply to that HTML file type/extension if varying outputs are needed.

Reply:
I don't see why a component would care if there's both an HTML and a CSS file associated with it. The only case I _think_ we care about is having two of the same filetype, like `foo.html.erb` and `foo.html.haml`, for example.

We also already have a way to determine the type of content a component should be rendering, via `format` https://github.com/ViewComponent/view_component/blob/5ae3c12e65628bf1d94d222737b0fd8ec0523b5d/lib/view_component/base.rb#L593-L595 (at the class level)

Reply:
@BlakeWilliams We're on the same page. The component should only care about what you mentioned. However the current functionality starts with the handler/processor and not the filetype such as `.html.*`.

Hence why there's a need to add a bit of code to limit the scope. This is to avoid the errors that arise if a component has more than one filetype per handler/processor.

Regarding the `format` at the class level you pointed out I did see that but I'm not certain it does much currently. However it doesn't mean that it shouldn't be leveraged in the solution we're aiming for. So tweaking things to use it is likely a good idea.